File: algorithms/find/linearSearch.ts
class LinearSearch {
    private process(arr: number[], v: number){
        for (let i = 0; i < arr.length; i++) {
            if(arr[i] === v) return true       
        }
        return false
    }

    public main(){
        const arr = [22,321,4,23,6,2,66,3245,755]
        const result = this.process(arr,66)
        if(result) console.log("Value find")
        else console.log("Value not find")
    }
}

(function(){
    new LinearSearch().main()
})()

File: golang/algos/find/linearSearch/LinearSearch.go
package linearsearch

import (
	"fmt"
	"log"
)

func LinearSearch(arr []int, v int) bool {
	for i := 0; i < len(arr); i++ {
		if arr[i] == v {
			return true
		}
	}
	return false
}

func Process() {
	arr := []int{22, 44, 11, 3421, 3254, 275, 975}
	err := LinearSearch(arr, 3254)
	if err != true {
		log.Fatal("Value not find")
	} else {
		fmt.Printf("Value find")
	}
}
